### ISSUE TITLE
CyFlash seems to hang at the Verify Row Ranges part of the bootload process

### ISSUE THREAD

**Issue by IronSean** [arachnidlabs/cyflash#10]
I'm running CyFlash (On both Python2 and Python3) over a Canbus interface. Between a linux computer with a can interface, and a Psoc device. 

I've used the example CANbus_Bootloader.c file to implement a basic canbus bootloader on our device which appears to run and begin communicating back. It properly enters the bootloader and returns data about the silicone ID, then it enters the "Verifying row ranges" section of processing and hangs. 

The stack trace I list might be off by a bit, as I reformatted some of the line spacing to conform with PEP-8 standards.

The stack trace is approximately:
bootload.py, line 303, in main
bootload.py, line 192, in bootload
bootload.py, line 2016, in verify_row_ranges
protocol.py, line 271, in get_flash_size
protocol.py, line 258, in send (which is BoorloaderSession.send)
protocol.py, line 268, in recv (which is CANbusTransport.recv)
socketcan, line52, recv:
[], []. timeout)[0]) > 0:

Looking on a CAN Analyzer, I see the following traffic:

6.344 s - Message 1(Enter Bootload?):
0x01, 0x38, 0x00, 0x00, 0xA0, 0x09, 0x17

6.345 s - Response(Silicone ID, Silicone Rev, Bootload Rev?):
0x01, 0x00, 0x08, 0x00, 0x69, 0x30, 0x12, 0x2E
0x00, 0x32, 0x01, 0x01, 0x55, 0x62, 0x17

6.418 s - Message 2(get Flash Size?):
0x01, 0x32, 0x01, 0x00, 0x00, 0xEB, 0x6B, 0x17

So it seems that the Bootloader is not replying to this second message requesting it's size.

**Comment by IronSean** [comment]
I found the source of the problem:

Line 278 of CANbus_Bootloader.c had the line `#ifdef CY_PSOC3` and then performed an endian swap for the PSOC3 architecture. However, the creator must have meant `#if CY_PSOC3` because CY_PSOC3 is still defined on PSOC5, it just evaluates to false.

After this, it causes another problem I through I saw with the Python2-3 compatibility, but wan't to confirm before trying to fix:

In Python2 calls to `super()` require the class to be passed. In Python3 `super()` is acceptable syntax and the class is implied automatically. This prevents the latest pulled code from running on Python 2. I'll try and figure out some more of these issues before forming another pull request.
